playbook/app/pb_kits/playbook/pb_kit/dateTime.ts

```diff
@@ -4,8 +4,19 @@ const days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
 
 const months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
 
+const formatDate = (newDate: Date | string) => {
+  const isTimelessStringDate = typeof newDate === "string" && !newDate.includes("T")
+
+  if (isTimelessStringDate) {
+    const unhyphenatedDate = new Date((newDate as string).replace(/-/g, "/"))
+    return unhyphenatedDate
+  }
+
+  return new Date(newDate)
+}
+
```

Comment:
Format the date only if it's a `string` AND does not contain a time. This is necessary because it could be in `ISO` format already (ex. `"2012-08-02T17:49:29Z"` as shown in the [Time Range Inline](https://playbook.powerapp.cloud/kits/time_range_inline/react) example).